jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Connection.java

```diff
@@ -231,6 +231,76 @@ private Runnable pollTask()
         }
     }
 
+    @Override
+    public void onHeaders(HeadersFrame frame)
+    {
+        session.onHeaders(frame);
+    }
+
+    @Override
+    public void onData(DataFrame frame)
+    {
+        NetworkBuffer networkBuffer = producer.networkBuffer;
+        networkBuffer.retain();
+        Callback callback = networkBuffer;
+        session.onData(frame, callback);
```

Comment:
It is needed because the `DataFrame` implicitly contains a slice of the `networkBuffer`.
This is the first available spot that we have to retain the `networkBuffer` before it is given to applications along with the callback to release it.
Remember that this code is older than the introduction of `Retainable` -- in Jetty 12 we do it differently and fully leverage `Retainable`.

Reply:
Can you add a short comment to that effect.